src/serverless-openapi-typescript.ts

```diff
@@ -2,241 +2,267 @@ import type Serverless from "serverless";
-      {
-        documentation: {
-          models: [{ name: modelName, contentType: 'application/json', schema: this.generateSchema(modelName) }]
+    }
+
+    setParamsDocumentation(params, httpEvent, documentationKey, functionName) {
+        Object.entries(params).forEach(([name, required]) => {
+            httpEvent.documentation[documentationKey] = httpEvent.documentation[documentationKey] || [];
+
+            const documentedParams = httpEvent.documentation[documentationKey];
+            const existingDocumentedParam = documentedParams.find(documentedParam => documentedParam.name === name);
+
+            if (existingDocumentedParam && typeof existingDocumentedParam.schema === 'string') {
+                existingDocumentedParam.schema = this.generateSchema(existingDocumentedParam.schema);
+            }
+
+            const getParamSchema = () => {
+                const definitionPrefix = this.getDefinitionPrefix(functionName);
+                const generatedSchemas = this.serverless.service.custom.documentation.models
+                const paramTypeName = `${definitionPrefix}.Request.${upperFirst(documentationKey)}.${name}`;
+                const haveSchema = !isEmpty(generatedSchemas.find(schemaName => schemaName.name === paramTypeName));
+                return haveSchema ? {$ref: `#/components/schemas/${paramTypeName}`} : {type: 'string'}
+            }
+
+            const schema = getParamSchema();
+            const paramDocumentationFromSls = {
+                name,
+                required,
+                schema
+            };
+
+            if (!existingDocumentedParam) {
+                documentedParams.push(paramDocumentationFromSls);
+            } else {
+                Object.assign(paramDocumentationFromSls, existingDocumentedParam);
+                Object.assign(existingDocumentedParam, paramDocumentationFromSls);
+            }
+        });
+    }
+
+    setModels(httpEvent, functionName, querystrings = {}) {
+        const definitionPrefix = this.getDefinitionPrefix(functionName);
+        const method = httpEvent.method.toLowerCase();
+        switch (method) {
+            case 'delete':
+                set(httpEvent, 'documentation.methodResponses', [{statusCode: 204, responseModels: {}}]);
+                break;
+            case 'patch':
+            case 'put':
+            case 'post':
+                const requestModelName = `${definitionPrefix}.Request.Body`;
+                this.setModel(`${definitionPrefix}.Request.Body`);
+                set(httpEvent, 'documentation.requestModels', {'application/json': requestModelName});
+                set(httpEvent, 'documentation.requestBody', {description: ''});
+            // no-break;
+            case 'get':
+                const responseModelName = `${definitionPrefix}.Response`;
+                this.setModel(responseModelName);
+                this.setQueryStringsModel(querystrings, definitionPrefix);
```

Comment:
```suggestion
                this.overrideDefaultQueryStringsModelByQueryParamsType(querystrings, definitionPrefix);
```

Reply:
thats not the override part
its only the part of generate the type

Reply:
that's really confusing...